src/tests/end-to-end/utils/utils.cpp

```diff
@@ -4,6 +4,208 @@
+    simulation_->variables.retrieveResultsForThermalCluster<Variable::Economy::VCardProductionByDispatchablePlant>(&result, cluster);
+    return result;
+}
+
+
+// -----------------------
+// BC rhs configuration
+// -----------------------
+BCrhsConfig::BCrhsConfig(std::shared_ptr<BindingConstraint> BC, unsigned int nbOfTimeSeries)
+    : nbOfTimeSeries_(nbOfTimeSeries), BC_(BC)
+{
+    BC_->RHSTimeSeries().resize(nbOfTimeSeries_, 8760);
+}
+
+void BCrhsConfig::fillTimeSeriesWith(unsigned int TSnumber, double rhsValue)
+{
+    if (TSnumber >= nbOfTimeSeries_)
+    {
+        logs.fatal() << "BCrhsConfig : TS number must be < Nb of TS";
+        AntaresSolverEmergencyShutdown();
+    }
+    BC_->RHSTimeSeries().fillColumn(TSnumber, rhsValue);
+}
+
+
+// --------------------------------------
+// BC group TS number configuration
+// --------------------------------------
+BCgroupScenarioBuilder::BCgroupScenarioBuilder(Study::Ptr study, unsigned int nbYears)
+    : nbYears_(nbYears)
+
+{
+    rules_ = createScenarioRules(study);
+}
+
+void BCgroupScenarioBuilder::yearGetsTSnumber(std::string groupName, unsigned int year, unsigned int TSnumber)
+{
+    if (year >= nbYears_)
+    {
+        logs.fatal() << "BCgroupScenarioBuilder : year number must be < Nb of MC years";
+        AntaresSolverEmergencyShutdown();
+    }
+
+    rules_->binding_constraints.setData(groupName, year, TSnumber + 1);
+}
+
+
+// =====================
+// Simulation handler
+// =====================
+
+void SimulationHandler::create()
+{
+    study_->initializeRuntimeInfos();
+    addScratchpadToEachArea(study_);
+
+    simulation_ = std::make_shared<ISimulation<Economy>>(*study_,
+                                                         settings_,
+                                                         &nullDurationCollector_);
+    SIM_AllocationTableaux();
+}
+
+
+// =========================
+// Basic study builder
+// =========================
+StudyBuilder::StudyBuilder()
+{
+    // Make logs shrink to errors (and higher) only
+    logs.verbosityLevel = Logs::Verbosity::Error::level;
+
+    study = std::make_shared<Study>();
+    simulation = std::make_shared<SimulationHandler>(study);
+
+    initializeStudy(study);
+    output = std::make_shared<OutputRetriever>(simulation->get());
+}
+
+void StudyBuilder::simulationBetweenDays(const unsigned int firstDay, const unsigned int lastDay)
+{
+    study->parameters.simulationDays.first = firstDay;
+    study->parameters.simulationDays.end = lastDay;
+}
+
+void StudyBuilder::giveWeigthOnlyToYear(unsigned int year)
+{
+    // Set all years weight to zero
+    unsigned int nbYears = study->parameters.nbYears;
+    for (unsigned int y = 0; y < nbYears; y++)
+        study->parameters.setYearWeight(y, 0.);
+
+    // Set one year's weight to 1
+    study->parameters.setYearWeight(year, 1.);
+
+    // Activate playlist, otherwise previous sets won't have any effect
+    study->parameters.userPlaylist = true;
+}
```

Comment:
You're right. It would be faster and simpler.
The original problem is that we can only retrieve synthesis results, a year by year results being erased by results of next year. A way to get a year results  is give a weight to this year only.
But we can activate this year only.